File: src/1662-Check-If-Two-String-Arrays-are-Equivalent/check_if_two_string_arrays_are_equivalent.cpp
#include <iostream>
#include <vector>
#include <string>

class Solution {
public:
    bool arrayStringsAreEqual(std::vector<std::string>& word1, std::vector<std::string>& word2) {
        std::string str1, str2;

        for (const std::string& w : word1) {
            str1 += w;
        }

        for (const std::string& w : word2) {
            str2 += w;
        }

        return str1 == str2;
    }
};

int main() {
    std::vector<std::string> word1 = {"ab", "c"};
    std::vector<std::string> word2 = {"a", "bc"};

    Solution solution;
    std::cout << std::boolalpha << solution.arrayStringsAreEqual(word1, word2);

    return 0;
}

File: src/1662-Check-If-Two-String-Arrays-are-Equivalent/check_if_two_string_arrays_are_equivalent.js
/**
 * @param {string[]} word1
 * @param {string[]} word2
 * @return {boolean}
 */
var arrayStringsAreEqual = function(word1, word2) {
  let str1 = ""
  let str2 = ""

  for (let w of word1) {
    str1 += w
  }

  for (let w of word2) {
    str2 += w
  }

  return str1 === str2
};

let word1 = ["ab", "c"]
let word2 = ["a", "bc"]

console.log(solution.arrayStringsAreEqual(word1, word2))